addons/website_sale_stock/static/src/js/variant_mixin.js

```diff
@@ -61,6 +62,10 @@ VariantMixin._onChangeCombinationStock = function (ev, $parent, combination) {
         }
     }
 
+	// needed xml-side for formatting of remaining qty
+	combination.formatFloat = field_utils.format.float;
+	combination.isInteger = Number.isInteger;
```

Comment:
maybe we could do a new method that will format float or integer and not put complexity in template?

side note: line have too much indentation